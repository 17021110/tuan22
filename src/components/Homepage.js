import React, {useState} from 'react';
import PropTypes from 'prop-types';
import '../App.css';
import {ButtonGroup, Button, Modal} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import * as Icon from 'react-feather';
Homepage.propTypes = {
  remote: PropTypes.func,
};
Homepage.defaultProps = {
  remote: null,

};

function Homepage(props) {
  const [modalShow1, setModalShow1] = useState(false);
  const {remote} = props;
  return (
    <div className='col-1'
      style={{backgroundColor: '#1b001c',
        display: 'block',
        justifyContent: 'space-between',
        position: 'absolute',
        right: '0px'}} >
      <div className='col-1' style={{height: '900px'}}>
        <ButtonGroup style={{display: 'block'}}>
          <Button style={{height: '60px',
            backgroundColor: '#1b001c',
            borderRadius: '0px',
            border: '0px',
            color: '#d79f47'}}
          onClick={remote} >
            <Icon.RefreshCw />
            <div>
              <span>Clean</span>
            </div>
          </Button>
          <Button style={{height: '60px',
            backgroundColor: '#1b001c',
            borderRadius: '0px',
            border: '0px',
            color: '#d79f47'}}
          onClick={() => setModalShow1(true)}>
            <Icon.Settings />
            <div>
              <span>Setting</span>
            </div>
          </Button>
          <Modal
            show={modalShow1}
            onHide={() => setModalShow1(false)}
          >
            <Modal.Header style={{backgroundColor: '#fcce84', border: '0px'}}>
              <Modal.Title id="contained-modal-title-vcenter" >
                <div className='col-12'
                  style={{display: 'flex',
                    justifyContent: 'flex-start'}}>
                  <Button style={{height: '50px',
                    width: '50px',
                    backgroundColor: '#927be3',
                    alignItems: 'center',
                    justifyContent: 'center',
                    alignContent: 'center',
                    display: 'flex',
                    borderRadius: '5px',
                    marginLeft: '-35px',
                    color: 'white'}}
                  onClick={() => setModalShow1((modalShow1) => !modalShow1)}>
                    <Icon.XCircle></Icon.XCircle>
                  </Button>
                  <div className='col-10 mx-auto' >
                    <h4 style={{justifyContent: 'center',
                      alignContent: 'center',
                      display: 'flex'}}>SETTINGS</h4>
                  </div>
                </div>
              </Modal.Title>
            </Modal.Header>
            <Modal.Body style={{backgroundColor: '#f8c46b'}}>
              <ButtonGroup vertical className='col-12'>

                <div className='col-sm-10 mx-auto mb-2'
                  style={{justifyContent: 'space-between',
                    display: 'flex'}}>
                  <Button className='col-sm-3'>left</Button>
                  <Button className='col-sm-3'>left</Button>
                </div>
                <Button className="col-sm-10 mx-auto  mb-2"
                  variant="secondary"
                  size="lg">Middle</Button>
                <Button className="col-sm-10 mx-auto mb-2"
                  variant="secondary"
                  size="lg">Right</Button>
                <Button className="col-sm-10 mx-auto mb-2"
                  variant="secondary"
                  size="lg">Right</Button>

              </ButtonGroup>
            </Modal.Body>
            <Modal.Footer style={{backgroundColor: '#f8c46b', border: '0px'}}>

            </Modal.Footer>
          </Modal>

          <Button style={{height: '60px',
            backgroundColor: '#1b001c',
            borderRadius: '0px',
            border: '0px',
            color: '#d79f47'}}>
            <Icon.BookOpen />
            <div>
              <span>Coures</span>
            </div>
          </Button>
          <Button style={{height: '60px',
            backgroundColor: '#1b001c',
            borderRadius: '0px',
            border: '0px',
            color: '#d79f47'}}>
            <Icon.Globe />
            <div>
              <span>Hnitsd</span>
            </div>
          </Button>

        </ButtonGroup>
      </div>

    </div >
  );
};
export default Homepage;
