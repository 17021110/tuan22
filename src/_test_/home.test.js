import React from 'react';
import { render } from '@testing-library/react';

import Homepage from '../components/HomePage';
test('should render homepage', () =>{
  const wrapper = render(<Homepage/>);
  expect(wrapper.container).toMatchSnapshot();
})