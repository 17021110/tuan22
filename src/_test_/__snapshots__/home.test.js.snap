// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render homepage 1`] = `
<div>
  <div
    class="col-1"
    style="background-color: rgb(27, 0, 28); display: block; justify-content: space-between; position: absolute; right: 0px;"
  >
    <div
      class="col-1"
      style="height: 900px;"
    >
      <div
        class="btn-group"
        role="group"
        style="display: block;"
      >
        <button
          class="btn btn-primary"
          style="height: 60px; background-color: rgb(27, 0, 28); border-radius: 0px; border: 0px; color: rgb(215, 159, 71);"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="23 4 23 10 17 10"
            />
            <polyline
              points="1 20 1 14 7 14"
            />
            <path
              d="M3.51 9a9 9 0 0 1 14.85-3.36L23 10M1 14l4.64 4.36A9 9 0 0 0 20.49 15"
            />
          </svg>
          <div>
            <span>
              Clean
            </span>
          </div>
        </button>
        <button
          class="btn btn-primary"
          style="height: 60px; background-color: rgb(27, 0, 28); border-radius: 0px; border: 0px; color: rgb(215, 159, 71);"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="3"
            />
            <path
              d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
            />
          </svg>
          <div>
            <span>
              Setting
            </span>
          </div>
        </button>
        <button
          class="btn btn-primary"
          style="height: 60px; background-color: rgb(27, 0, 28); border-radius: 0px; border: 0px; color: rgb(215, 159, 71);"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 3h6a4 4 0 0 1 4 4v14a3 3 0 0 0-3-3H2z"
            />
            <path
              d="M22 3h-6a4 4 0 0 0-4 4v14a3 3 0 0 1 3-3h7z"
            />
          </svg>
          <div>
            <span>
              Coures
            </span>
          </div>
        </button>
        <button
          class="btn btn-primary"
          style="height: 60px; background-color: rgb(27, 0, 28); border-radius: 0px; border: 0px; color: rgb(215, 159, 71);"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="2"
              x2="22"
              y1="12"
              y2="12"
            />
            <path
              d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z"
            />
          </svg>
          <div>
            <span>
              Hnitsd
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
